### DISCUSSION TITLE
Paginate to a specific object / Get page of object

### DISCUSSION THREAD

**Discussion by luap2703** [prisma/prisma#19028] [Q&A]
Hey,

We are wondering what the best practice is to **paginate** to a **specific object**

Our use-case is as following:

- We have a classic chat app with querying batches of messages in desc order from our Prisma-mysql db
- We now want to implement the search feature to go directly to a specific message. 
- To accomplish that  we need to fetch until that specific message still remaining the pagination logic (For caching reasons)

Our idea would be to get the page of the specific message by returning the count of messages that have been sent after that specific one, then dividing it by the page size. **Is there an alternative and/or better approach?**

**Comment by nurul3101** [comment]
Hello @paulo2703 👋 

Have you considered [cursor-based pagination](https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination)? It can be more efficient and flexible compared to the offset-based pagination that you are planning to use.

When a user searches for a specific message, you can find that message in the database and use the message's unique property value as the cursor and then you can query the database to fetch a batch of messages before the cursor and after the cursor, limiting the number of results by the page size.

**Reply by nurul3101** [reply]
Let us know in case it helps or if you need any assitance.

**Reply by nurul3101** [reply]
@paulo2703 If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
